File: students/k3340/Khurs_Pavel/lab1/udp-client.ts
import * as dgram from 'dgram';

const client = dgram.createSocket('udp4');
const message = Buffer.from('Hello, server');

client.send(message, 41234, 'localhost', (err) => {
  if (err) {
    console.error('Ошибка при отправке сообщения:', err);
    client.close();
  } else {
    console.log('Сообщение отправлено серверу');
  }
});

client.on('message', (msg) => {
  console.log(`Ответ от сервера: ${msg}`);
  client.close();
});

client.on('error', (err) => {
  console.error(`Ошибка клиента:\n${err.stack}`);
  client.close();
});

File: students/k3339/Skirlyak_Yaroslav/laboratory_work_1/1/udp_client.go
package main

import (
	"fmt"
	"net"
)

const (
	clientMessage = "Hello, server"
)

func main() {
	conn, err := net.Dial("udp", "127.0.0.1:8080")
	if err != nil {
		fmt.Printf("failed to dial: %v", err)
	}
	defer conn.Close()

	_, err = conn.Write([]byte(clientMessage))
	if err != nil {
		fmt.Printf("failed to write: %v", err)
	}

	buffer := make([]byte, 1024)
	_, err = conn.Read(buffer)
	if err != nil {
		fmt.Printf("failed to read: %v", err)
	}

	fmt.Printf("message: %s", buffer)
}
